src/main/java/it/pagopa/ecommerce/payment/methods/controller/PaymentMethodsController.java

```diff
@@ -183,4 +183,12 @@ public Mono<ResponseEntity<CalculateFeeResponseDto>> calculateFees(
                 ResponseEntity::ok
         );
     }
+
+    @Override
+    public Mono<ResponseEntity<PreauthorizationResponseDto>> createPreauthorization(
+                                                                                    String id,
+                                                                                    ServerWebExchange exchange
+    ) {
+        return Mono.just(ResponseEntity.status(HttpStatus.METHOD_NOT_ALLOWED).build());
```

Comment:
🚲 maybe we can use a not implemented yet error code here?

Reply:
[Looked into that](https://stackoverflow.com/questions/32889983/is-http-501-appropriate-for-an-unimplemented-api) 🙂
`501 Not Implemented` should be used only when a method is not recognized by the server indipendently from the resource, while `405 Method Not Allowed` is the correct HTTP status code if a request cannot be made to a resource with a method (technically the endpoint should always return an `Allow` header with a list of allowed methods when using `405`, but given that this implementation won't ever go to production I think it's safe to not be 100% compliant)